seismostats/seismicity/catalog.py

```diff
@@ -205,10 +221,140 @@ def bin_magnitudes(
             df = self.copy()
 
         df['magnitude'] = bin_to_precision(df["magnitude"], delta_m)
+        df.delta_m = delta_m
 
         if not inplace:
             return df
 
+    @require_cols(require=['magnitude'])
+    def estimate_mc(
+        self,
+        mcs_test: list | None = None,
+        delta_m: float | None = None,
+        p_pass: float = 0.05,
+        stop_when_passed: bool = True,
+        verbose: bool = False,
+        beta: float | None = None,
+        n_samples: int = 10000
+    ) -> tuple[np.ndarray, list[float], np.ndarray, float | None, float | None]:
+        """
+        Estimate the completeness magnitude (mc), possible mc values given as
+        an argument, or set to a range of values between min and max magnitude
+        in the catalog.
+
+        Args:
+            mcs_test:           Completeness magnitudes to test
+            delta_m:            Magnitude bins (sample has to be rounded to bins
+                            beforehand)
+            p_pass:             P-value with which the test is passed
+            stop_when_passed:   Stop calculations when first mc passes the test,
+                            by default True
+            verbose:            Verbose output, by default False
+            beta:               If beta is 'known', only estimate mc,
+                            by default None
+            n_samples:          Number of magnitude samples to be generated in
+                            p-value calculation of KS distance, default 10000
+
+        Returns:
+            mcs_test:   tested completeness magnitudes
+            ks_ds:      KS distances
+            ps:         p-values
+            best_mc:    best mc
+            beta:       corresponding best beta
+        """
+        if delta_m is None:
+            if self.delta_m is None:
+                raise ValueError("binning (delta_m) needs to be set")
+            else:
+                delta_m = self.delta_m
+
+        if mcs_test is None:
+            mcs_test = np.arange(self.magnitude.min(),
+                                 self.magnitude.max(),
+                                 delta_m)
+
+        mc_est = estimate_mc(self.magnitude,
```

Comment:
there will be an overall estimate_function soon, for now, it is probably the best to use the mc_maxcurvature as the standard, as it requires less knowledge.